packages/picasso.js/src/core/chart-components/labels/strategies/bars.js

```diff
@@ -32,9 +32,29 @@ function toBackground(label) {
   };
 }
 
-export function isTextInRect(rect, textMetrics, opts) {
-  return opts.rotate ? !(rect.width < textMetrics.height || rect.height < textMetrics.width)
-    : !(rect.width < textMetrics.width || rect.height < textMetrics.height);
+function isFitWidth(rect, label, rotate, ignoreCheck) {
+  return ignoreCheck || (rotate ? rect.width >= label.height : rect.width >= label.width);
+}
+
+function isFitHeight(rect, label, rotate, ignoreCheck) {
+  return ignoreCheck || (rotate ? rect.height >= label.width : rect.height >= label.height);
+}
+
+function isTextFitRect(orientation, rect, label, fitsHorizontally, overflow) {
+  let fitWidth;
+  let fitHeight;
+  if (orientation === 'v') {
+    fitWidth = fitsHorizontally || isFitWidth(rect, label, !fitsHorizontally, overflow);
```

Comment:
I don't think it adds any value to pass in the `overflow` variable to "isFitXXXX". It would be easier to read if it parts of this statement instead. Such that `fitWidth = fitsHorizontally || overflow || isFitWidth(rect, label, !fitsHorizontally);`

It would also skip one computation when the overflow parameter is not passed in to the function.

Reply:
Fixed!